src/utils/formatUtils.ts

```diff
@@ -109,3 +122,41 @@ export const getReferencedNodeInFormat = ({
 
   return null;
 };
+
+export const findReferencedNodeInText = ({
+  text,
+  discourseNode,
+}: {
+  text: string;
+  discourseNode: DiscourseNode;
+}) => {
+  // assumes that the referenced node in format has a specification
+  // which includes:
+  // has title relation
+  // a (.*?) pattern in it's target
+  // eg: Source: /^@(.*?)$/
+
+  const specification = discourseNode.specification;
+  const titleCondition = specification.find(
+    (s): s is QBClause => s.type === "clause" && s.relation === "has title"
+  );
+  if (!titleCondition) return null;
+
+  // Remove leading and trailing slashes and start/end modifiers
+  const patternStr = titleCondition.target.slice(1, -1).replace(/^\^|\$$/g, "");
+
+  // Since we aummar there's always a (.*?), we replace it with a specific pattern to capture text within [[ ]]
```

Comment:
> aummar

Reply:
I don't always `aummar`, but when I do, I prefer to leave it in a comment. 😅